build.sh

```diff
@@ -0,0 +1,239 @@
+#! /bin/bash
+msBuild='/c/Windows/Microsoft.NET/Framework64/v4.0.30319/'
+outputFolder='./_output'
+outputFolderMono='./_output_mono'
+outputFolderOsx='./_output_osx'
+outputFolderOsxApp='./_output_osx_app'
+testPackageFolder='./_tests/'
+testSearchPattern='*.Test/bin/x86/Release/*'
+sourceFolder='./src'
+updateFolder=$outputFolder/NzbDrone.Update
+updateFolderMono=$outputFolderMono/NzbDrone.Update
+
+CheckExitCode()
+{
+    "$@"
+    local status=$?
+    if [ $status -ne 0 ]; then
+        echo "error with $1" >&2
+        exit 1
+    fi
+    return $status
+}
+
+CleanFolder()
+{
+    local path=$1
+    local keepConfigFiles=$2
+
+    echo "Removing XMLDoc files"
+    find $path -name "*.xml" -delete
+    find $path -name "*.transform" -delete
```

Comment:
I've swapped it to use exec rm. I'm getting some warnings when it deletes
empty directories - I'll have a look tomorrow

On Thu, 23 Jul 2015 20:33 Mark McDowall <notifications@github.com> wrote:

> In build.sh
> <https://github.com/Sonarr/Sonarr/pull/661#discussion_r35361805>:
>
> > +    local status=$?
> > +    if [ $status -ne 0 ]; then
> > +        echo "error with $1" >&2
> > +        exit 1
> > +    fi
> > +    return $status
> > +}
> > +
> > +CleanFolder()
> > +{
> > +    local path=$1
> > +    local keepConfigFiles=$2
> > +
> > +    echo "Removing XMLDoc files"
> > +    find $path -name "*.xml" -delete
> > +    find $path -name "*.transform" -delete
>
> The -delete flag on find isn't supported by all systems, most do, but I
> don't think we can expect it to work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sonarr/Sonarr/pull/661/files#r35361805>.
>